### ISSUE TITLE
Plans 2023: biannual toggle - show 2yr cost savings in a discount manner

### ISSUE THREAD

**Issue by chriskmnds** [Automattic/wp-calypso#74261]
Part of addressing pdgrnI-28f-p2 

Update the prices shown in the grid to highlight the annual vs biannual difference, as per design.

### Media

![Image](https://user-images.githubusercontent.com/1705499/224006399-1f4544dd-bc55-4e70-8634-463c487b2192.png)

**Comment by chriskmnds** [comment]
> Show the cost saving by paying in a longer term in a discount manner

@southp concerning the cost saving, is there any consideration about these prices being discounted already, or do we always show the raw prices? I'm not sure how discounts work, but I'm guessing the cost saving should be "discounted yearly price" vs "discounted 2-yearly price" if there's a discount (coupon?) involved?

**Comment by southp** [comment]
> @southp concerning the cost saving, is there any consideration about these prices being discounted already, or do we always show the raw prices? I'm not sure how discounts work, but I'm guessing the cost saving should be "discounted yearly price" vs "discounted 2-yearly price" if there's a discount (coupon?) involved?

Sorry that I missed this ping, and thanks for raising the discussion in the team hangout.

In summary, we do have permanent discounts for specific currencies, e.g. INR, PHP, MXN. In those cases, let's use the discounted prices for comparison. 

Also, discounts incurred by coupons will only appear in checkout, so we don't need to consider that here.

**Comment by chriskmnds** [comment]
> > @southp concerning the cost saving, is there any consideration about these prices being discounted already, or do we always show the raw prices? I'm not sure how discounts work, but I'm guessing the cost saving should be "discounted yearly price" vs "discounted 2-yearly price" if there's a discount (coupon?) involved?
> 
> Sorry that I missed this ping, and thanks for raising the discussion in the team hangout.
> 
> In summary, we do have permanent discounts for specific currencies, e.g. INR, PHP, MXN. In those cases, let's use the discounted prices for comparison.
> 
> Also, discounts incurred by coupons will only appear in checkout, so we don't need to consider that here.

Thanks, @southp!

I think there's more to this than the currency discounts. In `trunk` if you view `/plans` with a paid plan, there's discounting being shown. I recall seeing this at times but couldn't pinpoint it. My guess is this is "upgrade" or monthly-vs-yearly related, coming from one of the selectors (e.g. [getPlanDiscountedRawPrice](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/sites/plans/selectors/get-plan-discounted-raw-price.js)). This is with a site on a Business plan:

<img width="700" alt="Screenshot 2023-03-15 at 1 35 35 PM" src="https://user-images.githubusercontent.com/1705499/225298099-ea49cece-99f9-4d77-8226-9a1cc487d3ec.png">

Bottom line, our selectors and grids prioritise discounting pricing, irrespective of where it's coming from. The setup is _"if there's a discounted price, show it, and cross out the raw price"_. So currency or not-currency related, I think we should account for it. We have the handling of that done in https://github.com/Automattic/wp-calypso/pull/74375, it's nearly done, although there are edge cases (and they may be API/non-ui related). I'll push that through for review soon to start fiddling with these.

**Comment by southp** [comment]
I'm moving this back to In Progress, since what we've agreed on [the accompanied PR](https://github.com/Automattic/wp-calypso/pull/74375) is that we will reshape it into just the code structure improvement for the discount displaying logics. We might want to just close this issue since the objective has changed; just that I didn't make up my mind at the moment.

**Comment by chriskmnds** [comment]
> I'm moving this back to In Progress, since what we've agreed on https://github.com/Automattic/wp-calypso/pull/74375 is that we will reshape it into just the code structure improvement for the discount displaying logics. We might want to just close this issue since the objective has changed; just that I didn't make up my mind at the moment.

The refactors have migrated to https://github.com/Automattic/wp-calypso/pull/74960 and we'll use it as a base for the price display updates :-)

**Comment by southp** [comment]
Given p2-pbxNRc-2qr#comment-4418, we won't do 1y/2y toggle in the forseeable future, so I'm also closing this since it's a related feature.
